testutil/testutil.go

```diff
@@ -0,0 +1,98 @@
+package testutil
+
+import (
+	"net"
+	"sync"
+)
+
+// Test constants
+const (
+	CloseOnMsgIdx = 5
+	MaxMsgCount   = 10
+)
+
+// LocalTCPServer tcp server for testing specific network errors
+type LocalTCPServer struct {
+	lnmu     sync.RWMutex
+	MockConn *Conn
```

Comment:
We should be able to. And specific errors are specific cases for specific tests, which can minimally mock what's necessary.